internal/pipeline/step.go

```diff
@@ -47,13 +48,19 @@ func (c *CommandStep) MarshalJSON() ([]byte, error) {
 			out[k] = v
 		}
 	}
+
 	if c.Command != "" {
 		out["command"] = c.Command
 	}
+
 	if len(c.Plugins) > 0 {
 		out["plugins"] = c.Plugins
 	}
 
+	if c.Signature != nil {
```

Comment:
Haha, whoops. 

🤔 It might be worth a tiny bit of reflection in a helper func that could be used across different structs, to cover the gap between yaml's `inline` feature and json's lack of it...

Reply:
yeah, i wondered about this as well... i might give it a crack just for fun

Reply:
okay, i've had my fun in 9962f55, if it seems like it's overcomplicating things (it might be?) i can just yeet that commit